internal/core/v1beta1/distribution_service.go

```diff
@@ -0,0 +1,150 @@
+package v1beta1
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/kanopy-platform/cdnvalidator/internal/config"
+	"github.com/kanopy-platform/cdnvalidator/internal/core"
+	"github.com/kanopy-platform/cdnvalidator/pkg/aws/cloudfront"
+)
+
+type DistributionService struct {
+	config     *config.Config
+	configFile string
+	cloudfront *cloudfront.Client
+	awsRegion  string
+	awsKey     string
+	awsSecret  string
+	timeout    time.Duration
+}
+
+func New(opts ...Option) (*DistributionService, error) {
+	var err error
+
+	d := &DistributionService{
+		config:  config.New(),
+		timeout: 30 * time.Second,
+	}
+
+	for _, opt := range opts {
+		opt(d)
+	}
+
+	// set up config
+	if d.configFile != "" {
+		if err := d.config.Watch(d.configFile); err != nil {
+			return nil, err
+		}
+	}
+
+	// set up AWS Cloudfront client
+	cfOpts := []cloudfront.Option{
+		cloudfront.WithAwsRegion(d.awsRegion),
+		cloudfront.WithStaticCredentials(d.awsKey, d.awsSecret),
+		cloudfront.WithTimeout(d.timeout),
+	}
+
+	d.cloudfront, err = cloudfront.New(cfOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return d, nil
+}
+
+func (d *DistributionService) getDistribution(ctx context.Context, distributionName string) (*config.Distribution, error) {
+	claims := core.GetClaims(ctx)
+	if len(claims) == 0 {
+		return nil, errors.New("no claims present")
+	}
+
+	// check user is entitled to the distributionName
+	entitlements := d.config.DistributionsFromClaims(claims)
```

Comment:
nit: these are actually `distributions`, I mean the concept of entitlements is somewhat hidden in config in some sense